### ISSUE TITLE
Docker-in-Docker failing in prow jobs

### ISSUE THREAD

**Issue by mboersma** [kubernetes-sigs/cluster-api-provider-azure#4553]
/kind bug

**What steps did you take and what happened:**

PRs recently began failing to set up Docker-in-Docker for prow jobs.

```
Unpacking azure-cli (2.57.0-1~bookworm) ...
Setting up azure-cli (2.57.0-1~bookworm) ...
WARNING: You may want to use 'az acr login -n capzci --expose-token' to get an access token, which does not require Docker to be installed.
ERROR: 2024-02-08 16:04:31.990310 An error occurred: DOCKER_COMMAND_ERROR
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

Problem occurs here:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/8f27cbeaa6d15c50d0af5494ec75fe5005562584/hack/ensure-acr-login.sh#L36

**What did you expect to happen:**

**Anything else you would like to add:**

Could be this? https://github.com/kubernetes/test-infra/pull/31873

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by mboersma** [comment]
See kubernetes/test-infra#31889

**Comment by mboersma** [comment]
/close

Fixed by kubernetes/test-infra#31900
